blueocean-dashboard/src/main/js/creation/bitbucket/cloud/BbCloudFlowManager.js

```diff
@@ -302,8 +302,21 @@ export default class BbCloudFlowManager extends FlowManager {
     _createPipelineComplete(result) {
         this.outcome = result.outcome;
         if (result.outcome === CreateMbpOutcome.SUCCESS) {
-            this.changeState(STATE.STEP_COMPLETE_SUCCESS);
-            this.pipeline = result.pipeline;
+            if (!this.isStateAdded(STATE.STEP_COMPLETE_MISSING_JENKINSFILE)) {
+                setTimeout(() => this._checkForBranchCreation(result.pipeline.name, true, ({ isFound, hasError, pipeline }) => {
+                    if (!hasError && isFound) {
+                        this._finishListening(STATE.STEP_COMPLETE_SUCCESS);
+                        this.pipeline = pipeline;
+                        this.pipelineName = pipeline.name;
+                    }
+                }), this.redirectTimeout);
+                if (!this.isStateAdded(STATE.STEP_COMPLETE_MISSING_JENKINSFILE)
+                    && !this.isStateAdded(STATE.STEP_COMPLETE_SUCCESS)) {
+                    this.changeState(STATE.PENDING_CREATION_SAVING);
+                    this.pipeline = result.pipeline;
+                    this.pipelineName = result.pipeline.name;
+                }
+            }
```

Comment:
> This code block is a bit confusing to me. The request to save completes (presumably w/ status=201). In what situation would the "missing jenkinsfile" state already be added? 

indexing build could be successful but it could have no jenkinsfile across branches - this information is known to us thru sse events. This block of code in certain cases get executed earlier than that. 

>Also I assumed after a successful save it would enter PENDING_CREATION_EVENTS rather than PENDING_CREATION_SAVING - as that state is set when the initial save is started.

I think this is fine.

`PENDING_CREATION_SAVING` - mbp pipeline was just created, we are waiting for sse events to tell us whether there was at least on pipeline created.

`PENDING_CREATION_EVENTS` - We got atleast one pipeline created - we are good to proceed to activity screen. It gets set when sse events related to pipeline build is received.

`PENDING_CREATION_EVENTS`, it waits and then redirects to activity page. `PENDING_CREATION_SAVING` is set when we receive pipeline builds later on.